QModManager/API/ModLoading/Internal/PatchMethod.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace QModManager.API.ModLoading.Internal
+{
+    using System;
+    using System.Reflection;
+    using QModManager.Utility;
+
+    internal class PatchMethod
+    {
+        public PatchMethod(MethodInfo method, IQModLoadable qmod)
+        {
+            this.ModId = qmod.Id;
+            this.Method = method;
+            this.LoadedAssembly = qmod.LoadedAssembly;
+        }
+
+        internal string ModId { get; }
+        internal MethodInfo Method { get; }
+        internal Assembly LoadedAssembly { get; }
+        internal bool IsPatched { get; private set; }
+
+        internal bool TryInvoke()
+        {
+            try
+            {
+                this.Method.Invoke(this.LoadedAssembly, new object[] { });
```

Comment:
This currently only works for `static` methods.  
With a little extra work, this and the supporting classes could be expanded to support non-static methods as well.

@AlexejheroYTB Do you see any benefit allowing this?  
The main benefit I see is the added flexibility would make it harder for first-time modders to "get it wrong".

Reply:
Non-static method handling still pending

Reply:
```csharp
var instance = Activator.CreateInstance(this.Method.DeclaringType);
this.Method.Invoke(instance, null);
```